### DISCUSSION TITLE
How to force HTTPS for next js website in production hosted on custom server?

### DISCUSSION THREAD

**Discussion by innocentinnox** [vercel/next.js#32615] [Help] [answered]
Am currently hosting my next js websites on my VPS with control web panel (CWP) installed on. I have an SSL installed and when I use the https URL it works fine but when I use an http URL it will stay in with the insecure connection. 

I've been looking for the solution for over 3 months and with the new next js 12 middleware I tried modifying all requests with http protocol to https but it keeps making an endless loop even if I used an if statement for only http URLs to redirect...

I also tried adding an htacess files with rules to force https since I have apace Nginx server but still not forcing https. 

I request your help guys to get it done.

**Comment by supermancash** [comment, marked as answer]
Hi Everyone coming back to this problem,

check out this [post](https://stackoverflow.com/questions/66458059/how-do-i-force-or-redirect-my-next-js-website-to-use-https) for a simple solution. 
IMPORTANT: 
The code snippet in the first answer hasn't been updated to use `${req.headers.get('host')}` instead of `req.nextUrl.hostname`
